### DISCUSSION TITLE
The current frame after `sly-eval-buffer` and `sly-compile-and-load-file`

### DISCUSSION THREAD

**Discussion by hww** [joaotavora/sly#519] [Q&A] [answered]
Which values control the current frame (cursor focus) after debugger popped up.

As I can see in the doom emacs  the `sly-eval-buffer` does not change current frame if debugger popped up
But with vanilla emacs the `sly-compile-and-load-file` does not change current frame if debugger popped up

So how I can make equal behavior for debugger in any emacs configuration?

**Comment by joaotavora** [comment, marked as answer]
evaluating and compiling code usually happens in a separate thread, and has no relation to whether a debugger is active or not.  I don't understand the differences between doom and vanilla emacs, but I don't use doom.  Maybe you could explain what you are trying to do.

**Reply by hww** [reply]
In the doom emacs the function `sly-eval-buffer` does not switch to debugger frame -- the focus stays in the source buffer. I want to make it work correctly.

**Reply by joaotavora** [reply]
I don't understand the problem, sorry. Are you evaluating a buffer that has a Lisp expression that causes an error when evaluated?  If so, then I think if doom emacs and vanilla emacs are not doing the right thing, then the fault is with doom emacs, or with your configuration.  I don't use either of those, so maybe someone else can help you.

**Reply by hww** [reply]
With doom emacs, evaluating the buffer with error will open debugger frame, but keep focus on source code (see image below). 
I believe the `doom` configuration (which by default uses `sly`) did not set correctly a value which control this behavior.  If there is no such value in the `sly` then there is a deeper problem with `doom` emacs. And I should avoid to use it.

With same setup the `slime` works correctly -- it change focus to the debugger.

![sly-doom](https://user-images.githubusercontent.com/773541/178252035-b89fdf19-8b69-4ab8-8583-3165a04dda01.png)

**Reply by joaotavora** [reply]
These values are generally in Emacs, not Sly.  But, again, I don't know exactly what you are doing because you have not described exactly.  And exact descriptions matter.  In other words, if there is indeed a problem and the problem is *not* in your configuration, you should do the work of coming up with a minimal reproducible example (see https://stackoverflow.com/help/minimal-reproducible-example) so that others can witness the problem by themselves, analyze it, and recommend fixes.  Else we're just wasting each other's time here.

**Reply by hww** [reply]
Ok. I will switch to the `slime` then. Thank You.

**Reply by joaotavora** [reply]
yes, switch to the slime, and good the luck

**Comment by monnier** [comment]
> In the Doom Emacs the function `sly-eval-buffer` does not switch to debugger
> frame -- the focus stays in the source buffer. I want to make it
> work correctly. 

Frame focus is a notoriously messy part of Emacs's behavior.  In large
part this is because it's not fully under the control of Emacs but
depends also on the window manager.

You should probably report this as a bug in Doom Emacs as well: where
the bug lies is not obvious at all based on the info you give, and we'll
need someone familiar with Doom (and/or its Sly config, if any) to help
find the origin of the problem.

**Reply by hww** [reply]
Ok. Thank You. I will report it at the DOOM project team. N.P.

**Reply by joaotavora** [reply]
I'm almost certain that it's not a frame focus issue, but rather window selection issue.  SLY has window selection code, but without a reproduction recipe, we can't know what's happening. As far as I understand everything works ok in "vanilla Emacs".

**Reply by hww** [reply]
Right, there are no problems with vanilla.
